### ISSUE TITLE
Advanced queries do not work after installing ItemSetsTree

### ISSUE THREAD

**Issue by pprw** [biblibre/omeka-s-module-ItemSetsTree#2]
After installing this module, I cannot anymore make an advanced query.

Here is stack trace, when trying to do so:

```
Omeka S a rencontré une erreur

Doctrine\ORM\Query\QueryException
[Syntax Error] line 0, col 72: Error: Expected Doctrine\ORM\Query\Lexer::T_IDENTIFIER, got '0'

Détails :

Doctrine\ORM\Query\QueryException: SELECT omeka_root FROM Omeka\Entity\ItemSet omeka_root WHERE omeka_root.0 = :omeka_0 in /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/QueryException.php:43
Stack trace:
#0 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(457): Doctrine\ORM\Query\QueryException::dqlError('SELECT omeka_ro...')
#1 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(325): Doctrine\ORM\Query\Parser->syntaxError('Doctrine\\ORM\\Qu...')
#2 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1084): Doctrine\ORM\Query\Parser->match(102)
#3 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1130): Doctrine\ORM\Query\Parser->PathExpression(12)
#4 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2844): Doctrine\ORM\Query\Parser->SingleValuedPathExpression()
#5 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2801): Doctrine\ORM\Query\Parser->ArithmeticPrimary()
#6 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2769): Doctrine\ORM\Query\Parser->ArithmeticFactor()
#7 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2743): Doctrine\ORM\Query\Parser->ArithmeticTerm()
#8 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2730): Doctrine\ORM\Query\Parser->SimpleArithmeticExpression()
#9 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(3063): Doctrine\ORM\Query\Parser->ArithmeticExpression()
#10 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2594): Doctrine\ORM\Query\Parser->ComparisonExpression()
#11 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2475): Doctrine\ORM\Query\Parser->SimpleConditionalExpression()
#12 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2451): Doctrine\ORM\Query\Parser->ConditionalPrimary()
#13 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2419): Doctrine\ORM\Query\Parser->ConditionalFactor()
#14 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2394): Doctrine\ORM\Query\Parser->ConditionalTerm()
#15 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1361): Doctrine\ORM\Query\Parser->ConditionalExpression()
#16 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(881): Doctrine\ORM\Query\Parser->WhereClause()
#17 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(848): Doctrine\ORM\Query\Parser->SelectStatement()
#18 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(261): Doctrine\ORM\Query\Parser->QueryLanguage()
#19 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(360): Doctrine\ORM\Query\Parser->getAST()
#20 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(286): Doctrine\ORM\Query\Parser->parse()
#21 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(298): Doctrine\ORM\Query->_parse()
#22 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(992): Doctrine\ORM\Query->_doExecute()
#23 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(947): Doctrine\ORM\AbstractQuery->executeIgnoreQueryCache(NULL, NULL)
#24 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(789): Doctrine\ORM\AbstractQuery->execute(NULL, NULL)
#25 /sites/ahponcar/www/una2gm/application/src/Api/Adapter/AbstractEntityAdapter.php(625): Doctrine\ORM\AbstractQuery->getOneOrNullResult()
#26 /sites/ahponcar/www/una2gm/application/src/Api/Adapter/AbstractEntityAdapter.php(354): Omeka\Api\Adapter\AbstractEntityAdapter->findEntity(Array, Object(Omeka\Api\Request))
#27 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(230): Omeka\Api\Adapter\AbstractEntityAdapter->read(Object(Omeka\Api\Request))
#28 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(115): Omeka\Api\Manager->execute(Object(Omeka\Api\Request))
#29 /sites/ahponcar/www/una2gm/modules/ItemSetsTree/Module.php(263): Omeka\Api\Manager->read('item_sets', Array)
#30 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): ItemSetsTree\Module->onItemApiSearchPre(Object(Zend\EventManager\Event))
#31 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(171): Zend\EventManager\EventManager->triggerListeners(Object(Zend\EventManager\Event))
#32 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(300): Zend\EventManager\EventManager->triggerEvent(Object(Zend\EventManager\Event))
#33 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(216): Omeka\Api\Manager->initialize(Object(Omeka\Api\Adapter\ItemAdapter), Object(Omeka\Api\Request))
#34 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(59): Omeka\Api\Manager->execute(Object(Omeka\Api\Request))
#35 /sites/ahponcar/www/una2gm/modules/Search/src/Api/ManagerDelegator.php(35): Omeka\Api\Manager->search('items', Array, Array)
#36 /sites/ahponcar/www/una2gm/application/src/Mvc/Controller/Plugin/Api.php(63): Search\Api\ManagerDelegator->search('items', Array, Array)
#37 /sites/ahponcar/www/una2gm/application/src/Controller/Admin/ItemController.php(37): Omeka\Mvc\Controller\Plugin\Api->search('items', Array)
#38 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php(78): Omeka\Controller\Admin\ItemController->browseAction()
#39 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#40 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#41 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(104): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#42 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/DispatchListener.php(138): Zend\Mvc\Controller\AbstractController->dispatch(Object(Zend\Http\PhpEnvironment\Request), Object(Zend\Http\PhpEnvironment\Response))
#43 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#44 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#45 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/Application.php(332): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#46 /sites/ahponcar/www/una2gm/index.php(21): Zend\Mvc\Application->run()
#47 {main}

Next Doctrine\ORM\Query\QueryException: [Syntax Error] line 0, col 72: Error: Expected Doctrine\ORM\Query\Lexer::T_IDENTIFIER, got '0' in /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/QueryException.php:54
Stack trace:
#0 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(457): Doctrine\ORM\Query\QueryException::syntaxError('line 0, col 72:...', Object(Doctrine\ORM\Query\QueryException))
#1 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(325): Doctrine\ORM\Query\Parser->syntaxError('Doctrine\\ORM\\Qu...')
#2 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1084): Doctrine\ORM\Query\Parser->match(102)
#3 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1130): Doctrine\ORM\Query\Parser->PathExpression(12)
#4 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2844): Doctrine\ORM\Query\Parser->SingleValuedPathExpression()
#5 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2801): Doctrine\ORM\Query\Parser->ArithmeticPrimary()
#6 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2769): Doctrine\ORM\Query\Parser->ArithmeticFactor()
#7 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2743): Doctrine\ORM\Query\Parser->ArithmeticTerm()
#8 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2730): Doctrine\ORM\Query\Parser->SimpleArithmeticExpression()
#9 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(3063): Doctrine\ORM\Query\Parser->ArithmeticExpression()
#10 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2594): Doctrine\ORM\Query\Parser->ComparisonExpression()
#11 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2475): Doctrine\ORM\Query\Parser->SimpleConditionalExpression()
#12 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2451): Doctrine\ORM\Query\Parser->ConditionalPrimary()
#13 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2419): Doctrine\ORM\Query\Parser->ConditionalFactor()
#14 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2394): Doctrine\ORM\Query\Parser->ConditionalTerm()
#15 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1361): Doctrine\ORM\Query\Parser->ConditionalExpression()
#16 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(881): Doctrine\ORM\Query\Parser->WhereClause()
#17 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(848): Doctrine\ORM\Query\Parser->SelectStatement()
#18 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(261): Doctrine\ORM\Query\Parser->QueryLanguage()
#19 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(360): Doctrine\ORM\Query\Parser->getAST()
#20 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(286): Doctrine\ORM\Query\Parser->parse()
#21 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(298): Doctrine\ORM\Query->_parse()
#22 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(992): Doctrine\ORM\Query->_doExecute()
#23 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(947): Doctrine\ORM\AbstractQuery->executeIgnoreQueryCache(NULL, NULL)
#24 /sites/ahponcar/www/una2gm/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(789): Doctrine\ORM\AbstractQuery->execute(NULL, NULL)
#25 /sites/ahponcar/www/una2gm/application/src/Api/Adapter/AbstractEntityAdapter.php(625): Doctrine\ORM\AbstractQuery->getOneOrNullResult()
#26 /sites/ahponcar/www/una2gm/application/src/Api/Adapter/AbstractEntityAdapter.php(354): Omeka\Api\Adapter\AbstractEntityAdapter->findEntity(Array, Object(Omeka\Api\Request))
#27 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(230): Omeka\Api\Adapter\AbstractEntityAdapter->read(Object(Omeka\Api\Request))
#28 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(115): Omeka\Api\Manager->execute(Object(Omeka\Api\Request))
#29 /sites/ahponcar/www/una2gm/modules/ItemSetsTree/Module.php(263): Omeka\Api\Manager->read('item_sets', Array)
#30 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): ItemSetsTree\Module->onItemApiSearchPre(Object(Zend\EventManager\Event))
#31 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(171): Zend\EventManager\EventManager->triggerListeners(Object(Zend\EventManager\Event))
#32 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(300): Zend\EventManager\EventManager->triggerEvent(Object(Zend\EventManager\Event))
#33 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(216): Omeka\Api\Manager->initialize(Object(Omeka\Api\Adapter\ItemAdapter), Object(Omeka\Api\Request))
#34 /sites/ahponcar/www/una2gm/application/src/Api/Manager.php(59): Omeka\Api\Manager->execute(Object(Omeka\Api\Request))
#35 /sites/ahponcar/www/una2gm/modules/Search/src/Api/ManagerDelegator.php(35): Omeka\Api\Manager->search('items', Array, Array)
#36 /sites/ahponcar/www/una2gm/application/src/Mvc/Controller/Plugin/Api.php(63): Search\Api\ManagerDelegator->search('items', Array, Array)
#37 /sites/ahponcar/www/una2gm/application/src/Controller/Admin/ItemController.php(37): Omeka\Mvc\Controller\Plugin\Api->search('items', Array)
#38 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php(78): Omeka\Controller\Admin\ItemController->browseAction()
#39 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#40 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#41 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(104): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#42 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/DispatchListener.php(138): Zend\Mvc\Controller\AbstractController->dispatch(Object(Zend\Http\PhpEnvironment\Request), Object(Zend\Http\PhpEnvironment\Response))
#43 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#44 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#45 /sites/ahponcar/www/una2gm/vendor/zendframework/zend-mvc/src/Application.php(332): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#46 /sites/ahponcar/www/una2gm/index.php(21): Zend\Mvc\Application->run()
#47 {main}

```

**Comment by jajm** [comment]
Hi,
Thanks for reporting the bug, this should be fixed by d06af5a1b98cacb26489cdfd0386d5e347bfa5c6
Can you get the latest master and confirm that the problem is fixed ?

**Comment by pprw** [comment]
Problem fixed indeed! Thanks
